stl/inc/xmemory

```diff
@@ -1033,147 +1034,190 @@ struct _Iterator_base0 {
             }
+        }
+    }
 
-            _STL_VERIFY(*_Pnext, "ITERATOR LIST CORRUPTED!");
-            *_Pnext  = _Mynextiter;
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
+    _CONSTEXPR20_DYNALLOC void _Adopt(const _Container_base12* _Parent) noexcept {
+        if (_Parent) { // have a parent, do adoption
+            _Myproxy = _Parent->_Myproxy;
+        } else { // no future parent, just disown current parent
             _Myproxy = nullptr;
         }
     }
-#endif // _ITERATOR_DEBUG_LEVEL == 2
+#endif // _ITERATOR_DEBUG_LEVEL != 2
+
+    _CONSTEXPR20_DYNALLOC const _Container_base12* _Getcont() const noexcept {
+        return _Myproxy ? _Myproxy->_Mycont : nullptr;
+    }
 
     static constexpr bool _Unwrap_when_unverified = _ITERATOR_DEBUG_LEVEL == 0;
 
-    _Container_proxy* _Myproxy;
-    _Iterator_base12* _Mynextiter;
-};
+    _Container_proxy* _Myproxy    = nullptr;
+    _Iterator_base12* _Mynextiter = nullptr;
 
-// MEMBER FUNCTIONS FOR _Container_base12
-inline void _Container_base12::_Orphan_all() noexcept {
 #if _ITERATOR_DEBUG_LEVEL == 2
-    if (_Myproxy) { // proxy allocated, drain it
+private:
+    _CONSTEXPR20_DYNALLOC void _Adopt_unlocked(_Container_proxy* _Parent_proxy) noexcept {
+        if (_Myproxy) { // adopted, remove self from list
+            _Orphan_me_unlocked();
+        }
+        _Mynextiter                 = _Parent_proxy->_Myfirstiter;
+        _Parent_proxy->_Myfirstiter = this;
+        _Myproxy                    = _Parent_proxy;
+    }
+
+    void _Adopt_locked(_Container_proxy* _Parent_proxy) noexcept {
         _Lockit _Lock(_LOCK_DEBUG);
+        _Adopt_unlocked(_Parent_proxy);
+    }
 
-        for (auto _Pnext = &_Myproxy->_Myfirstiter; *_Pnext; *_Pnext = (*_Pnext)->_Mynextiter) {
-            (*_Pnext)->_Myproxy = nullptr;
+    _CONSTEXPR20_DYNALLOC void _Orphan_me_unlocked() noexcept {
+        _Iterator_base12** _Pnext = &_Myproxy->_Myfirstiter;
+        while (*_Pnext && *_Pnext != this) {
+            _Pnext = &(*_Pnext)->_Mynextiter;
         }
 
-        _Myproxy->_Myfirstiter = nullptr;
+        _STL_VERIFY(*_Pnext, "ITERATOR LIST CORRUPTED!");
+        *_Pnext  = _Mynextiter;
+        _Myproxy = nullptr;
+    }
+
+    void _Orphan_me_locked() noexcept {
+        _Lockit _Lock(_LOCK_DEBUG);
+        _Orphan_me_unlocked();
     }
 #endif // _ITERATOR_DEBUG_LEVEL == 2
+};
+
+// MEMBER FUNCTIONS FOR _Container_base12
+_CONSTEXPR20_DYNALLOC void _Container_base12::_Orphan_all_unlocked() noexcept {
+    for (auto _Pnext = &_Myproxy->_Myfirstiter; *_Pnext; *_Pnext = (*_Pnext)->_Mynextiter) {
+        (*_Pnext)->_Myproxy = nullptr;
+    }
+    _Myproxy->_Myfirstiter = nullptr;
 }
 
-inline void _Container_base12::_Swap_proxy_and_iterators(_Container_base12& _Right) noexcept {
-#if _ITERATOR_DEBUG_LEVEL == 2
+inline void _Container_base12::_Orphan_all_locked() noexcept {
     _Lockit _Lock(_LOCK_DEBUG);
+    _Orphan_all_unlocked();
+}

```

Comment:
I believe this could simply be defined within the `_Container_base12` class definition, as it doesn't directly mention `_Iterator_base12` machinery. (We conventionally avoid out-of-line member function definitions except when they are absolutely necessary.)